### ISSUE TITLE
[bug] [@eslint-react/prefer-read-only-props] false positive using React types

### ISSUE THREAD

**Issue by danielrentz** [Rel1cx/eslint-react#962]
### Describe the bug

Rule `@eslint-react/prefer-read-only-props` triggers when using `Readonly<React.HTMLAttributes<T>>`

### Reproduction

https://stackblitz.com/edit/vitejs-vite-znmyb7tb

> npm run lint

### Expected behavior

No type errors

### Platform and versions

```Text
plugin 1.30.0
```

### Stack trace

```Text

```
